var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -112,6 +112,12 @@ class VtkM(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("hip@3.7:", when="+rocm")
 
+    conflicts("+ascent_types", when="+64bitids", msg="Ascent types requires 32 bit IDs")
+    conflicts("+logging", when="@:1.3.0", msg="logging is not supported in vtk-m<1.3")
+    conflicts("+mpi", when="@:1.3.0", msg="mpi is not supported in vtk-m<1.3")
+    conflicts("+openmp", when="@:1.3.0", msg="mpi is not supported in vtk-m<1.3")
+    conflicts("+testlib", when="@:1.7.0", msg="testlib is not supported in vtk-m<1.7")
```

Comment:
Most of these (all maybe?) can be incorporated into the corresponding variants as a when clause.